src/iuse.cpp

```diff
@@ -954,11 +954,12 @@ int iuse::flusleep(player *p, item *it, bool, const tripoint& )
 
 int iuse::inhaler(player *p, item *it, bool, const tripoint& )
 {
-    p->remove_effect( effect_asthma);
-    p->add_msg_if_player(m_neutral, _("You take a puff from your inhaler."));
-    if (one_in(50)) {  // adverse reaction
-        p->add_msg_if_player(m_bad, _("Your heart begins to race."));
-        p->mod_fatigue(-10);
+    p->add_msg_if_player( m_neutral, _( "You take a puff from your inhaler." ) );
+    if( !p->remove_effect( effect_asthma) ) {
+        p->mod_fatigue( -10 ); // if we don't have asthma can be used as stimulant
```

Comment:
If I read it correctly, you significantly increase its stimulating effect from some minor chance of fatigue reduction (```one_in(50)```) to guaranteed one. I'm able to decrease level of fatigue from "dead tired" to OK condition by 30 puffs. I'm not sure it's intended/good change.

We can add some more significant penalties for overdosing, including hallucinations, vomiting, and so on.